x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -542,13 +534,8 @@ void decodeToken(String token, ActionListener<UserToken> listener) {
                     return;
                 }
                 final String accessToken = in.readString();
-                // TODO Remove this conditional after backporting to 7.x
-                if (version.onOrAfter(VERSION_HASHED_TOKENS)) {
```

Comment:
This one was truly pointless: this check here is already under the "if" condition `version.onOrAfter(VERSION_ACCESS_TOKENS_AS_UUIDS)` and `VERSION_HASHED_TOKENS` == `VERSION_ACCESS_TOKENS_AS_UUIDS` == `7.12` .